### ISSUE TITLE
E1101 error on classes with __getattr__

### ISSUE THREAD

**Issue by xobs** [PyCQA/pylint#2331]
### Steps to reproduce
1. Create a Python file with the following code:
```python
import collections
isinstance([], collections.Iterable)
```
2. Run pylint on that file

### Current behavior

`pylint` reports `deleteme.py:2:15: E1101: Module 'collections' has no 'Iterable' member (no-member)`

### Expected behavior

`pylint` shouldn't report an error, since the member exists because Python does not exit, and `collections` contains a `__getattr__` parameter.

### pylint --version output

```sh
D:\> pylint --version
pylint 2.0.0
astroid 2.0.1
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]
```

### Other notes

I am investigating this issue because I'm receiving a lot of these errors when using `migen`.  It is reporting very many `E1101` errors for objects that are derived from a base `Module` class that implements `__getattr__()`.  I also noticed this particular error for `collections`, which is why I've reported that error here -- it requires no additional packages, and should be reproducible on a stock Python build.

**Comment by PCManticore** [comment]
Thanks for the report! The best way to deal with these kind of modules is to simply ignore them from the `no-member` check as there is no way to figure out statically if a member exist or not. We do the same thing for classes that implements `__getattr__` or `__getattribute__`. It's kind of a shame really as this might mean we could miss potential bugs in user code, but it's also a shame that we have to resort to module `__getattr__` for minor functionality.

**Comment by xobs** [comment]
Thanks!  How can I test this?  I just did a `git clone git@github.com:PyCQA/pylint.git`, followed by a `cd pylint` and then a `python setup.py install`.  According to `Get-Command pylint.exe | Get-ChildItem`, the global pylint.exe was modified (i.e. LastWriteTime is set to now).

`pylint--version` prints out 2.0.0:

```sh
[2:13:23 PM] D:/Code/pylint> pylint --version
pylint 2.0.0
astroid 2.0.1
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]
[2:13:27 PM] D:/Code/pylint> git rev-parse HEAD
820d3c82fc4f52645f64789e29be55bd8b57a114
[2:13:29 PM] D:/Code/pylint>
```

I'm still seeing E1101 errors in modules that use `__getattr__`

**Comment by PCManticore** [comment]
@xobs This seems to suggest that it either uses the wrong version of `pylint` or that it wasn't installed properly. Can you check if the `pylint` you are running also has the same commit? I cannot reproduce the problem locally.

**Comment by xobs** [comment]
I still get the same error when running the version of `pylint` present in the `bin/` directory:

```sh
[2:53:56 PM] D:/Code/pylint> type .\deleteme.py
import collections
isinstance([], collections.Iterable)
[2:53:57 PM] D:/Code/pylint> git rev-parse HEAD
820d3c82fc4f52645f64789e29be55bd8b57a114
[2:54:05 PM] D:/Code/pylint> python bin\pylint --version
pylint 2.0.0
astroid 2.0.1
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]
[2:54:09 PM] D:/Code/pylint> python bin\pylint .\deleteme.py
************* Module deleteme
deleteme.py:2:15: E1101: Module 'collections' has no 'Iterable' member (no-member)

----------------------------------------------------------------------
Your code has been rated at -15.00/10 (previous run: -15.00/10, +0.00)

[2:54:22 PM] D:/Code/pylint>
```
